routing/dht/routing.go

```diff
@@ -170,7 +168,7 @@ func (dht *IpfsDHT) FindProviders(ctx context.Context, key u.Key) ([]peer.PeerIn
 // to the given key
 func (dht *IpfsDHT) getClosestPeers(ctx context.Context, key u.Key) (<-chan peer.ID, error) {
 	e := log.EventBegin(ctx, "getClosestPeers", &key)
-	tablepeers := dht.routingTable.NearestPeers(kb.ConvertKey(key), AlphaValue)
+	tablepeers := dht.routingTable.ListPeers()
```

Comment:
Yeah, i know kademlia isnt this. I posted in another issue (i cant find it) about s/kademlia and their disjoint path lookups. see 4.4 in http://www.tm.uka.de/doc/SKademlia_2007.pdf even they arent as aggressive as kicking off the query with the entire routing table. This is not a nice thing to do in general, and we should move as fast as we can to using the s/kademlia closest (but disjoint path) lookups. But the fact remains that our dht currently **_does not_** resolve queries properly. getting it to work using all the information available will be a good start, from there we can constrain it.